File: src/programmers/Solution_캐시.java
package programmers;

import java.util.LinkedList;
import java.util.Queue;

public class Solution_캐시 {
	public int solution(int cacheSize, String[] cities) {
        int answer = 0;
        if(cacheSize==0) {
        	return cities.length*5;
        }
        
        Queue<String> queue =  new LinkedList<>();
        
        for(String city: cities) {
                city = city.toLowerCase();
        	if (queue.contains(city)) {
        		answer += 1;
        		queue.remove(city);
        	}else {
        		if(queue.size()==cacheSize) {
        			queue.poll();
        		}
        		answer += 5;
        	}
        	queue.add(city);
        }
        
        return answer;
    }
}

File: src/programmers/Solution_캐시.js
function solution(cacheSize, cities) {
    var answer = 0;
    if(cacheSize===0){
        return cities.length*5;
    }
    
    let list = [];
    
    cities.forEach(function(city){
        city = city.toLowerCase();
        if(list.includes(city)){
            list.splice(list.indexOf(city),1);
            answer += 1;
        }else{
            if(list.length===cacheSize){
                list.splice(0,1);
            }
            answer += 5;
        }
        list.push(city);
    })
    return answer;
}